### PR TITLE
LUCENE 9204: Move SpanQuery and subclasses to the queries module

### PR DESCRIPTION
**Pull request by romseygeek**

### CODE REVIEW

File: lucene/core/src/test/org/apache/lucene/search/TestBooleanQuery.java [hunk 1]
```diff
@@ -386,41 +382,6 @@ public void testBS2DisjunctionNextVsAdvance() throws Exception {
     d.close();
   }
 
-  // LUCENE-4477 / LUCENE-4401:
```
Comment (romseygeek) [inline lucene/core/src/test/org/apache/lucene/search/TestBooleanQuery.java:389]: Moved to spans/TestBasics

File: lucene/core/src/test/org/apache/lucene/search/TestComplexExplanations.java [hunk 2]
```diff
@@ -43,126 +40,6 @@ public void tearDown() throws Exception {
     super.tearDown();
   }
 
-  public void test1() throws Exception {
```
Comment (romseygeek) [inline lucene/core/src/test/org/apache/lucene/search/TestComplexExplanations.java:46]: These have all moved to TestSpanExplanations

File: lucene/core/src/test/org/apache/lucene/search/TestDisjunctionMaxQuery.java [hunk 3]
```diff
@@ -457,40 +455,6 @@ public void testBooleanOptionalWithTiebreakerAndBoost() throws Exception {
     }
   }
 
-  // LUCENE-4477 / LUCENE-4401:
```
Comment (romseygeek) [inline lucene/core/src/test/org/apache/lucene/search/TestDisjunctionMaxQuery.java:460]: Moved to TestBasics

File: lucene/core/src/test/org/apache/lucene/search/TestMatchesIterator.java [hunk 4]
```diff
@@ -443,37 +439,6 @@ public void testExactMultiPhraseQuery() throws IOException {
   //  0         1         2         3         4         5         6         7
   // "a phrase sentence with many phrase sentence iterations of a phrase sentence",
 
-  public void testSpanQuery() throws IOException {
```
Comment (romseygeek) [inline lucene/core/src/test/org/apache/lucene/search/TestMatchesIterator.java:446]: Moved to TestSpanMatches

File: lucene/core/src/test/org/apache/lucene/search/TestPositionIncrement.java [hunk 5]
```diff
@@ -189,106 +177,4 @@ public void reset() throws IOException {
     reader.close();
     store.close();
   }
-
```
Comment (romseygeek) [inline lucene/core/src/test/org/apache/lucene/search/TestPositionIncrement.java:192]: Moved to TestPayloadSpanPositions

File: lucene/core/src/test/org/apache/lucene/search/TestQueryRescorer.java [hunk 6]
```diff
@@ -185,18 +182,6 @@ public void testBasic() throws Exception {
     assertEquals("1", searcher.doc(hits2.scoreDocs[0].doc).get("id"));
     assertEquals("0", searcher.doc(hits2.scoreDocs[1].doc).get("id"));
 
-    // Resort using SpanNearQuery:
```
Comment (romseygeek) [inline lucene/core/src/test/org/apache/lucene/search/TestQueryRescorer.java:188]: Moved to TestQueryRescorerWithSpans

File: lucene/core/src/test/org/apache/lucene/search/TestQueryVisitor.java [hunk 7]
```diff
@@ -56,15 +53,6 @@
                       .build(),
                   3),
               BooleanClause.Occur.MUST)
```
Comment (romseygeek) [inline lucene/core/src/test/org/apache/lucene/search/TestQueryVisitor.java:55]: Moved to TestSpanQueryVisitor

File: lucene/core/src/test/org/apache/lucene/search/similarities/TestSimilarity2.java [hunk 8]
```diff
@@ -268,33 +265,4 @@ public void testOmitTFAndNorms() throws Exception {
     ir.close();
     dir.close();
   }
-
```
Comment (romseygeek) [inline lucene/core/src/test/org/apache/lucene/search/similarities/TestSimilarity2.java:271]: Moved to TestSpanSimilarity

**Review by mikemccand — APPROVED** [review]
Woot!  Was this just a rote move/fixup package paths?  Thanks @romseygeek!

**Review by romseygeek — COMMENTED** [review]
This is the big one!  The vast majority of this change is just package renames, but there's a bit of test shuffling as well.  I have commented on each test case to show where it has been moved to.

**Pull request merged**